Mathlib/Analysis/NormedSpace/Unitization.lean

```diff
@@ -0,0 +1,248 @@
+/-
+Copyright (c) 2023 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+
+import Mathlib.Algebra.Algebra.Unitization
+import Mathlib.Analysis.NormedSpace.OperatorNorm
+
+/-!
+# Unitization norms
+
+Given a not necessarily unital normed `𝕜`-algebra `A`, it is frequently of interest to equip its
+`Unitization` with a norm which simultaneously makes it into a normed algebra and also satisfies
+two properties:
+
+- `‖1‖ = 1` (i.e., `NormOneClass`)
+- The embedding of `A` in `Unitization 𝕜 A` is an isometry. (i.e., `Isometry Unitization.inr`)
+
+One way to do this is to equip it with the norm from `PiLp 1` (actually, it should be
+`ProdLp 1`, but that doesn't exist), that is, `‖(k, a)‖ = ‖k‖ + ‖a‖`. However, when the norm on `A`
+is *regular* (i.e., `ContinuousLinearMap.mul`) is an isometry, there is another natural choice:
+the pullback of the norm on `𝕜 × (A →L[𝕜] A)` under the map
+`(k, a) ↦ (k, k • 1 + ContinuousLinearMap.mul 𝕜 A a)`. It turns out that among all norms on the
+unitization satisfying the properties specified above, the norm inherited from `PiLp 1` is maximal,
+and the norm inherited from this pullback is minimal.
+
+For possibly non-unital `RegularNormedAlgebra`s  `A` (over `𝕜`), we construct a `NormedAlgebra`
+structure on `Unitization 𝕜 A` using the pullback described above. The reason for choosing this norm
+is that for a C⋆-algebra `A` its norm is always regular, and the pullback norm on `Unitization 𝕜 A`
+is then also a C⋆-norm.
+
+## Main definitions
+
+- `Unitization.splitMul : Unitization 𝕜 A →ₐ[𝕜] (𝕜 × (A →L[𝕜] A))`: The first coordinate of this
+  map is just `Unitization.fst` and the second is the `Unitization.lift` of the left regular
+  representation of `A` (i.e., `NonUnitalAlgHom.Lmul`). We use this map to pull back the
+  `NormedRing` and `NormedAlgebra` structures.
+
+## Main statements
+
+- `Unitization.instNormedRing`, `Unitization.instNormedAlgebra`, `Unitization.instNormOneClass`,
+  `Unitization.instCompleteSpace`: when `A` is a non-unital Banach `𝕜`-algebra with a regular norm,
+  then `Unitization 𝕜 A` is a unital Banach `𝕜`-algebra with `‖1‖ = 1`.
+- `Unitization.norm_inr`, `Unitization.isometry_inr`: the natural inclusion `A → Unitization 𝕜 A`
+  is an isometry, or in mathematical parlance, the norm on `A` extends to a norm on
+  `Unitization 𝕜 A`.
+
+## Implementation details
+
+We ensure that the uniform structure, and hence also the topological structure, is definitionally
+equal to the pullback of `instUniformSpaceProd` allong `Unitization.addEquiv` (this is essentially
```

Comment:
equal to the pullback of `instUniformSpaceProd` along `Unitization.addEquiv` (this is essentially